inspirehep/modules/workflows/tasks/matching.py

```diff
@@ -113,45 +107,35 @@ def article_exists(obj, eng):
     return False
 
 
-@with_debug_logging
-def belongs_to_relevant_category(obj, eng):
-    """Return True if the record is Core or Non-Core.
+def can_auto_approve(obj, eng):
+    """Check if the record in `obj.data` can be auto approved.
 
     Arguments:
         obj: a workflow object.
         eng: a workflow engine.
 
-    Returns:
-        bool: ``True`` if the ingested article belongs to a relevant
-        category, else ``False``.
-
-    """
-    return get_coreness(obj.data) is not Coreness.non_relevant
-
-
-def get_coreness(record):
-    """Return the coreness of the record harvested.
-
-    Args:
-        record(dict): the record harvested.
-
     Return:
-        a Coreness value in according to the record's coreness level.
+        True either the record belongs to one of `core` or `no-core` arxiv
+        categories, else False.
     """
-    arxiv_categories = set(get_arxiv_categories(record))
-    relevant_categories = current_app.config.get('ARXIV_CATEGORIES', [])
-
-    if arxiv_categories & set(relevant_categories['core']):
-        return Coreness.core
-    elif arxiv_categories & set(relevant_categories['non-core']):
-        return Coreness.non_core
-    else:
-        return Coreness.non_relevant
+    rec_cats = set(get_arxiv_categories(obj.data))
+    arxiv_categs = current_app.config.get('ARXIV_CATEGORIES', {})
+    return len(
+        rec_cats & set(arxiv_categs.get('core') + arxiv_categs.get('non-core'))
+    ) > 0
 
 
 @with_debug_logging
-def set_coreness_in_extra_data(obj, eng):
-    obj.extra_data['core'] = get_coreness(obj.data).value
+def set_core_in_extra_data(obj, eng):
+    """Set `core=True` in `obj.extra_data` if the record is core"""
+    def _is_core(record):
+        return len(
```

Comment:
the `len` thing is not needed, as empty sets are falsy and non-empty sets are truthy.